### PR TITLE
feat(libs): add libpng decoder

### PR DESCRIPTION
**Pull request by FASTSHIFT**
### Description of the feature or fix

Added [libpng](https://github.com/glennrp/libpng) porting, support ARM-NENO instruction accelerated decoding.

### Checkpoints
- [x] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [x] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [x] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [x] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [x] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [x] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [x] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [x] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [x] Use typed pointers instead of `void *` pointers
- [x] Do not `malloc` into a static or global variables. Instead declare the variable in `lv_global_t` structure in [`lv_global.h`](https://github.com/lvgl/lvgl/blob/master/src/core/lv_global.h) and mark the variable with `(LV_GLOBAL_DEFAULT()->variable)` when it's used. See a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [x] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [x] Widget members function must start with `lv_<modul_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [x] `struct`s should be used via an API and not modified directly via their elements.
- [x] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [x] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [x] Arguments must be named in H files too.
- [x] To register and use callbacks one of the following needs to be followed (see a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)):
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

File: Kconfig [hunk 1]
```diff
@@ -1023,6 +1023,9 @@ menu "LVGL configuration"
 		config LV_USE_PNG
```
Comment (kisvegabor) [inline Kconfig]: Please rename it to make it cleaner which PNG library is which. Also add a few lines of description to make it easier to decide which one to choose.
```suggestion
		config LV_USE_LODEPNG
```

File: src/libs/libpng/lv_libpng.c [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+        lv_cache_entry_t * cache = lv_cache_find(fn, LV_CACHE_SRC_TYPE_STR, 0, 0);
+        if(cache) {
+            dsc->img_data = lv_cache_get_data(cache);
+            dsc->user_data = cache;     /*Save the cache to release it in decoder_close*/
+            lv_cache_unlock();
+            return LV_RESULT_OK;
+        }
+    }
+
+    else if(dsc->src_type == LV_IMAGE_SRC_VARIABLE) {
```
Comment (kisvegabor) [inline src/libs/libpng/lv_libpng.c]: It seems in `decoder_open` only files are handled, so the `if(dsc->src_type == LV_IMAGE_SRC_VARIABLE)` shouldn't be required.

File: Kconfig (hunk 1 above)
Comment (FASTSHIFT) [reply Kconfig]: Can I submit another PR to rename `lv_png` -> `lv_lodepng`?

Comment (kisvegabor) [reply Kconfig]: Sure! Can I merge this one?

**Comment by kisvegabor** [comment]
Oh sorry, I forgot the "best" part: please add  basic docs page and at least one screenshot test.

File: docs/libs/libpng.rst [hunk 3]
```diff
@@ -0,0 +1,42 @@
+==============
+libpng decoder
+==============
+
+libpng is the official PNG reference library. It supports almost all PNG features, is extensible, and has been extensively tested for over 28 years.
+Detailed introduction: `libjpeg-turbo <http://www.libpng.org/pub/png/libpng.html>`__.
```
Comment (kisvegabor) [inline docs/libs/libpng.rst]:
```suggestion
Detailed introduction: `libpng <http://www.libpng.org/pub/png/libpng.html>`__.
```

**Comment by kisvegabor** [comment]
Looks good, let me know if it's ready to merge.

**Comment by FASTSHIFT** [comment]
> Looks good, let me know if it's ready to merge.

The test passes and we can merge it :)

**Comment by kisvegabor** [comment]
I've just realized that we need to modify the example like [this](https://github.com/lvgl/lvgl/blob/master/examples/libs/libjpeg_turbo/lv_example_libjpeg_turbo_1.c) because we don't compile libpng to WebAssembly for the docs, so we need a fallback. 

Please update the example, and I'll merge it immediately. :slightly_smiling_face:

**Comment by FASTSHIFT** [comment]
> I've just realized that we need to modify the example like [this](https://github.com/lvgl/lvgl/blob/master/examples/libs/libjpeg_turbo/lv_example_libjpeg_turbo_1.c) because we don't compile libpng to WebAssembly for the docs, so we need a fallback.
> 
> Please update the example, and I'll merge it immediately. 🙂

Done.

**Comment by kisvegabor** [comment]
Thank you, merging! :tada:

**Pull request merged**